### PR TITLE
Update UG & DG with statistics functionality

### PR DESCRIPTION
**Pull request by zenlyj**
* Update user guide and developer guides with stats functionality
* Create assertion statements for statistics functionality

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -168,6 +168,36 @@ according to an option and shows the updated list.
...
+* The current flashcard list is obtained from the `ModelManager`.
+
+
+* If a valid flashcard index was included in the user input, the flashcard
+identified by the provided index is retrieved from the current flashcard list, and the statistics associated with the card is obtained by
+`Flashcard#getStats()`.
+
+* If the flashcard index was omitted from the user input. A new `Statistics` object is created, representing the
+overall statistics of the current flashcard list.
+
```
Comment (vuminhhieunus2019) [inline docs/DeveloperGuide.md:190]: You can add which methods you implement for this stats command, you can refer to sort and undo/redo implementation for more details "It implements the following operations:" part.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -168,6 +168,36 @@ according to an option and shows the updated list.
...
+
+A `CommandResult` is created with the generated flashcard `Statistics`. It is then passed to `MainWindow`, where
+the UI is updated to display the retrieved statistics.
+
+The following sequence diagram illustrates the scenario when user enters `stats 3` into the command box and presses
+`enter`.
+
+![StatsSequenceDiagram](images/StatsSequenceDiagram.png)
+
```
Comment (vuminhhieunus2019) [inline docs/DeveloperGuide.md:198]: <div markdown="span" class="alert alert-info">:information_source: **Note:** The lifeline for `StatsCommandParser` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.

</div>
Add this note under the sequence diagram to indicate the limitation of PlantUML

File: docs/UserGuide.md [hunk 3]
```diff
@@ -287,6 +288,30 @@ When the user enters `review` in the command box, this new window will appear. <
...
+
+When user enters `stats 2` in the command box and presses `enter`, the statistics of the <br>
+2nd flashcard in the list is displayed.
+
+![UiReview](./images/UiStatsIndex.png) <br><br>
+
+When user enters `stats` in the command box and presses `enter`, the overall statistics of <br>
+the current flashcard list is displayed.
+
+![UiReview](./images/UiStatsIndex.png) <br><br>
```
Comment (vuminhhieunus2019) [inline docs/UserGuide.md]: image
I'm unable to see the image, you might want to check the link again

File: docs/UserGuide.md [hunk 4]
```diff
@@ -333,6 +358,18 @@ Format: `a`
 Hides the answer of the current flashcard. <br>
 Format: `h`
 
+### Review flashcard as correct : `t`
+
+Marks that the user got the answer correct for the current flashcard. <br>
+Note: This command can only be executed if the answer of the current flash card is shown. <br>
+Format: `t`
+
+### Review flashcard as wrong : `f`
```
Comment (vuminhhieunus2019) [inline docs/UserGuide.md:370]: Remember to add these 2 commands in the table of content.

**Review by vuminhhieunus2019 — COMMENTED** [review]
Overall, the documentation is quite good. Just take note and update the issues that I showed in my review.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (zenlyj) [reply docs/DeveloperGuide.md:190]: Great suggestion. I have updated the section accordingly.

File: docs/UserGuide.md (hunk 3 above)
Comment (zenlyj) [reply docs/UserGuide.md]: Fixed. It should show now.

File: docs/UserGuide.md (hunk 4 above)
Comment (zenlyj) [reply docs/UserGuide.md:370]: Added. Also fixed the broken links in the table of content.

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (zenlyj) [reply docs/DeveloperGuide.md:198]: Updated, thanks for mentioning!

**Review by yeoutzer — COMMENTED** [review]
LGTM.

**Pull request merged**